src/Abuse/Adapters/TimeLimit.php

```diff
@@ -149,55 +127,72 @@ protected function parseKey(): string
      */
     protected function count(string $key, int $time): int
     {
-        if(0 == $this->limit) { // No limit no point for counting
+        if (0 == $this->limit) { // No limit no point for counting
             return 0;
         }
 
-        if(!\is_null($this->count)) { // Get fetched result
+        if (!\is_null($this->count)) { // Get fetched result
             return $this->count;
         }
-                
-        $st = $this->getPDO()->prepare('SELECT _count FROM `' . $this->getNamespace() . '.abuse.abuse`
-          WHERE _key = :key AND _time = :time
-          LIMIT 1;
-		');
-
-        $st->bindValue(':key',     $key,    PDO::PARAM_STR);
-        $st->bindValue(':time',    $time,   PDO::PARAM_STR);
 
-        $st->execute();
-
-    	$result = $st->fetch();
-    	$result = (\is_array($result) && isset($result['_count'])) ? $result['_count'] : 0;
+        Authorization::disable();
+        $result = $this->db->find(TimeLimit::COLLECTION, [
+            new Query('_key', Query::TYPE_EQUAL, [$key]),
+            new Query('_time', Query::TYPE_EQUAL, [$time]),
+        ], 1);
+        Authorization::reset();
+
+        if (count($result) == 1) {
```

Comment:
Can we clearly distinguish between `Timelimit->count()` and PHP's standard `count()` and check strict equality:
```suggestion
        if (\count($result) === 1) {
```